core/plugins/org.polarsys.capella.core.data.la.ui.quickfix/src/org/polarsys/capella/core/data/la/ui/quickfix/resolver/LogicalComponentRealizedSystemComponentsResolver.java

```diff
@@ -42,21 +49,52 @@ public void run() {
           Project project = ProjectExt.getProject(modelElements.get(0));
           BlockArchitecture architecture = BlockArchitectureExt
               .getBlockArchitecture(CtxPackage.Literals.SYSTEM_ANALYSIS, project);
-          systemComponent = (SystemComponent) architecture.getSystem();
-          for (EObject obj : modelElements) {
-            if (obj instanceof LogicalComponent) {
-              logicalComponent = (LogicalComponent) obj;
+          if (architecture != null) {
+            systemComponent = (SystemComponent) architecture.getSystem();
+            for (EObject obj : modelElements) {
+              if (obj instanceof LogicalComponent) {
+                logicalComponent = (LogicalComponent) obj;
+              }
+            }
+            if (logicalComponent != null) {
+              EList<ComponentRealization> componentRealizationList = logicalComponent.getOwnedComponentRealizations();
+              ComponentRealization cr = null;
+              if (componentRealizationList != null && !componentRealizationList.isEmpty()) {

```

Comment:
FYI, componentRealizationList  can't be null (see org.polarsys.capella.core.data.cs.impl.ComponentImpl.getOwnedComponentRealizations())